### PR TITLE
Adapt behavior of NewMethodTransformation. NewMethodTransformations n…

### PR DESCRIPTION
**Pull request by legionth**
…eeds a concrete object instead just class name. According to input validation paper.

Shoud also fix the current warnings/error in the CI.

### CODE REVIEW

File: src/Refinery/To/Group.php [hunk 1]
```diff
@@ -165,6 +165,6 @@ public function toNew($classNameOrArray): Transformation
 	 */
 	public function data(string $dataType): Transformation
 	{
-		return $this->toNew(array('\ILIAS\Data\Factory', $dataType));
+		return $this->toNew(array(new \ILIAS\Data\Factory(), $dataType));
```
Comment (klees) [inline src/Refinery/To/Group.php]: Let's make this dependency explicit by using DI.

File: src/Refinery/To/Transformation/NewMethodTransformation.php [hunk 2]
```diff
@@ -17,38 +17,38 @@
 class NewMethodTransformation implements Transformation
 {
 	use DeriveApplyToFromTransform;
+
 	/**
 	 * @var
```
Comment (klees) [inline src/Refinery/To/Transformation/NewMethodTransformation.php]:
```suggestion
	 * @var object
```

File: src/Refinery/To/Transformation/NewMethodTransformation.php [hunk 3]
```diff
@@ -17,38 +17,38 @@
...
 	 */
-	private $className;
+	private $object;
 
 	/**
 	 * @var
```
Comment (klees) [inline src/Refinery/To/Transformation/NewMethodTransformation.php]:
```suggestion
	 * @var string
```

File: src/Refinery/To/Transformation/NewMethodTransformation.php [hunk 4]
```diff
@@ -17,38 +17,38 @@
...
 	 */
 	private $method;
 
 	/**
-	 * @param string $className
+	 * @param string $object
 	 * @param string $methodToCall
 	 */
-	public function __construct(string $className, string $methodToCall)
+	public function __construct($object, string $methodToCall)
```
Comment (klees) [inline src/Refinery/To/Transformation/NewMethodTransformation.php]:
```suggestion
	public function __construct(object $object, string $methodToCall)
```

File: src/Refinery/To/Transformation/NewMethodTransformation.php [hunk 5]
```diff
@@ -17,38 +17,38 @@
...
 	{
-		if (false === class_exists($className)) {
+		if (false === is_object($object)) {
 			throw new ConstraintViolationException(
 				'The first parameter MUST be an object',
 				'first_parameter_must_be_an_object'
 			);
 		}
```
Comment (klees) [inline src/Refinery/To/Transformation/NewMethodTransformation.php]: With the type-hint on `object` we can remove this completely.

File: src/Refinery/To/Transformation/NewMethodTransformation.php [hunk 6]
```diff
@@ -17,38 +17,38 @@
...
 
-
-		if (false === method_exists($className, $methodToCall)) {
+		if (false === method_exists($object, $methodToCall)) {
 			throw new ConstraintViolationException(
```
Comment (klees) [inline src/Refinery/To/Transformation/NewMethodTransformation.php]: Please use `\InvalidArgumentException` here. If the method does not exist on the object this does not mean that some input violated a constraint, but instead that some developer constructed a transformation improperly.

**Review by klees — CHANGES_REQUESTED** [review]
Hi @legionth,

thx for taking care of this. LGTM in general. I found some issue I missed in the original PR. Please fix them.

Best regards!

**Comment by legionth** [comment]
Thanks for the review :+1:
Implemented all the inquired points.

Ready to review.

If this looks good to you, I would like to check travis first. It seems there us currently a problem with the pull request script with travis. We may have to check manually.
@gvollbach is informed about this :)

File: Services/Init/classes/class.ilInitialisation.php [hunk 7]
```diff
@@ -1654,8 +1654,7 @@ protected static function initRefinery()
 	{
 		$container['refinery'] = function ($container) {
 			$dataFactory = new \ILIAS\Data\Factory();
-			$validationFactory = new \ILIAS\Refinery\Validation\Factory($dataFactory, $container->language());
-			return new \ILIAS\Refinery\Factory($validationFactory);
+			return new \ILIAS\Refinery\Factory($dataFactory);
```
Comment (mjansenDatabay) [inline Services/Init/classes/class.ilInitialisation.php:1657]: @legionth / @klees  : Should we put the `\ILIAS\Data\Factory` into the `DIC` as well? Or do we want to hide this from public and only provide implicit access/usage via the `Refinery` strategies. Any opinions?

**Comment by legionth** [comment]
Travis seems to work correctly now. Thx @gvollbach :+1: 

The build looks also fine :shipit:

File: Services/Init/classes/class.ilInitialisation.php (hunk 7 above)
Comment (klees) [reply Services/Init/classes/class.ilInitialisation.php:1657]: Isn't it in the `DIC` already? The UI-Framework uses it as well. If not, I would put it there. If people want to depend on it (like Refinery and UI-Framework) that's fine for me.

**Review by klees — APPROVED** [review]

File: Services/Init/classes/class.ilInitialisation.php (hunk 7 above)
Comment (mjansenDatabay) [reply Services/Init/classes/class.ilInitialisation.php:1657]: No, it's not, yet.

**Pull request merged**